impala/dbapi/__init__.py

```diff
@@ -33,21 +33,42 @@
 threadsafety = 1  # Threads may share the module, but not connections
 paramstyle = 'pyformat'
 
-
 def connect(host='localhost', port=21050, protocol='hiveserver2',
             database=None, timeout=45, use_ssl=False, ca_cert=None,
             use_ldap=False, ldap_user=None, ldap_password=None,
-            use_kerberos=False, kerberos_service_name='impala'):
+            use_kerberos=False, kerberos_service_name='impala',
+            auth_mechanism=None):
+
+    # Supported authentication mechanisms
+    auth_mechanisms = ['NOSASL', 'PLAIN', 'GSSAPI', 'LDAP']
+    if use_kerberos:
```

Comment:
I like how you eliminated the `use_kerberos` option in the other functions.  Do you think it's safe to do so here?  Or should we keep it for backwards compatibility?  Perhaps we check whether `use_kerberos=True`, and if so, emit a deprecation message and manually set `auth_mechanism='GSSAPI'?`